### PR TITLE
Avoid creating unnecessary exception instances on profile validation

### LINKED ISSUE #34848: Too many exceptions created when validating user profile
### Description

When validating a user profile, KC always creates an exception that is then only thrown when a validation error occurs, that is rare. 

### Discussion

_No response_

### Motivation

Creating exceptions with a stack trace is slow and creates an overhead. It is suspected to be extra slow on ARM.

### Details

I'll create a PR.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #34848

### CODE REVIEW

**Review by pruivo — APPROVED** [review]
It LGTM but I left some suggestions for code improvements. 
Also, the class `Error` does not make much sense to me since every error is a `ValidationError`; the map's value could be a list of `ValidationError` and save allocating an `Error` instance.

**Comment by ahus1** [comment]
@pruivo - reworked the PR, and applied almost all changes. Left out the refactoring about the errors. BTW, I think this in 99% of the time doesn't do anything as users are usually valid, so I wouldn't optimize it too much here.

File: server-spi-private/src/main/java/org/keycloak/userprofile/ValidationException.java [hunk 1]
```diff
@@ -35,15 +35,21 @@
...
-public final class ValidationException extends RuntimeException implements Consumer<ValidationError> {
+public final class ValidationException extends RuntimeException {
 
-	private final Map<String, List<Error>> errors = new HashMap<>();
+	private final Map<String, List<Error>> errors;
 
-	public ValidationException() {}
+	public ValidationException() {
+        errors = new HashMap<>();
+    }
```
Comment (pruivo) [inline server-spi-private/src/main/java/org/keycloak/userprofile/ValidationException.java:44]: The indentation seems a little off. 
can be an empty immutable map
```java
errors = Map.of(); 
```

**Comment by ahus1** [comment]
The indentation looked off as the original file used tabs and not spaces. This is now fixed, but you would need to use probably IntelliJ to see the real changes now.

**Review by pruivo — APPROVED** [review]
LGTM

**Review by mhajas — APPROVED** [review]
Thank you for the improvements @ahus1! The changes LGTM.

Flamegraph for client credentials grant
before:
image


after:
image


This improves time spent on signing from ~12% to ~16%.

Btw. together with the previous PR we are now at ~18% with flamegraph:
image

**Pull request merged**